run_common.go

```diff
@@ -1618,6 +1619,9 @@ func (b *Builder) getSecretMount(tokens []string, secrets map[string]define.Secr
 			id = kv[1]
 		case "target", "dst", "destination":
 			target = kv[1]
+			if !path.IsAbs(target) {
+				target = filepath.Join(workdir, target)
```

Comment:
Do we have to worry about this path escaping the container?
../../../../etc/passwd or something like that

Reply:
I don't think it can escape container , I tried this 
```Dockerfile
FROM docker.io/ubuntu:22.04

WORKDIR /somedir

RUN --mount=type=secret,id=secret-foo,dst=secret1.txt --mount=type=secret,id=secret-bar,dst=../../../etc/passwd cat /etc/passwd
```
```
./buildah build --no-cache --secret id=secret-foo,src=build/secret1.txt  -t test1 .
```

Reply:
Yeah, that feels twitch to me.  I see the build succeeded, but did it get the hosts passed file in the image?

Reply:
@TomSweeneyRedHat No hosts passed file secret will not be part of the image unleass manual `cp` is made, any file or secret added by `--mount` instruction should not persist after run instruction.

Reply:
Ok maybe this is a misunderstanding on my part, but does this effect the destination not the source.  I am less concerned about the source then the destination.

Can I do something like

RUN --mount=type=secret,id=secret-foo,dst=../../../secret1.txt --mount=type=secret,id=secret-bar,dst=../../../secret2.txt \

Reply:
Yes this allowed and secret gets created on the path related to the `workdir` and does not persists after the `RUN` statement.

Reply:
`filepath.Join()` returns a `Clean()`ed path, and `Clean()` removes any initial "/.." components.